### DISCUSSION TITLE
Does pylance support displaying a url with a label?

### DISCUSSION THREAD

**Discussion by shadycuz** [microsoft/pylance-release#1160] [Q&A]
I'm not sure if label is the correct terminology.  But you know when you have [some word](https://google.com) that is backed by a URL. 

pylance seems to support URLs but not the label kind, I have tried different variations including [google](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html) and [reST](https://docs.typo3.org/m/typo3/docs-how-to-document/master/en-us/WritingReST/Hyperlinks.html) style links. 

I opened a discussion because I wasn't sure if I was just doing it wrong or maybe its the responsibility of the python extension and not the language server.

![image](https://user-images.githubusercontent.com/17090999/114852325-47fec680-9e00-11eb-8ea3-df5ee2e92607.png)

Docstring:

```
"""AWS Intrinsic functions.

This module contains the logic to render both AWS `Intrinsic function`_
and https://google.com

by the `Google Python Style Guide`_.
.. _Intrinsic function:
   https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/intrinsic-function-reference.html
.. _Google Python Style Guide:
   http://google.github.io/styleguide/pyguide.html
"""
```

**Comment by jakebailey** [comment]
Can you file an issue for this? This'd be an enhancement.

Our docstring parser is very loose and tries to understand most formats; this is a ReST specific format that we don't handle.

**Reply by shadycuz** [reply]
I can do that, but is there a format you do handle? Other than just pasting in a URL I could not get one with a label to work.

**Reply by jakebailey** [reply]
https://github.com/microsoft/pylance-release/issues/new/choose and just fill out the feature request template.

**Reply by gvanrossum** [reply]
> is there a format you do handle? Other than just pasting in a URL I could not get one with a label to work.

Perhaps the Markdown format works? That would be `[label](url)`.

**Reply by shadycuz** [reply]
> Perhaps the Markdown format works? That would be `[label](url)`.

Even after reading the source code of pyright and seeing that the extension processed Markdown, I still didn't think of this 😊.

Didn't work for me though.

![image](https://user-images.githubusercontent.com/17090999/114990643-590a0f00-9eb6-11eb-88b6-6b36ad5625b8.png)

**Reply by jakebailey** [reply]
No, I'm afraid none of these methods will work; the docstring converter is only a loose converter, and doesn't actually parse or understand any given docstring format. It will treat this as regular text depending on the context. To really support formats, we will need to do something else, but that's trickier to manage when everyone has their own formats.

**Reply by shadycuz** [reply]
This seems like a good use case for hooks? I've never developed a plugin for vscode before so I'm not sure what it is capable of but what if pylance DID NOT implement this feature because of all the different formats but instead offered to allow other plugins to render the docstrings for pylance. That would allow someone to develop a plugin for reST that implements all those features. This leaves pylance to focus only whats in the PEP specification for docstrings.
